src/transcode.jl

```diff
@@ -76,9 +94,29 @@ julia> String(decompressed)
 
 ```
 """
-function Base.transcode(codec::Codec, data::ByteData)
-    input = Buffer(data)
-    output = Buffer(initial_output_size(codec, buffermem(input)))
+function Base.transcode(
+    codec::Codec,
+    input::Buffer,
+    output::Union{Buffer,Nothing} = nothing,
+)
+    output = (output === nothing ? _default_output_buffer(codec, input) : initbufer!(output))
+    transcode!(output, codec, input)
+end
+
+"""
+    transcode!(output::Buffer, codec::Codec, input::Buffer)
```

Comment:
What do you think of the order of the arguments here? By convention `output` should go first as it is being modified.

If `Base.transcode` did not have `codec` first, I would have probably elected to put it at the end. I ended up with the current order to stay consistent with `Base.transcode`.

Reply:
I personally think putting output first seems reasonable, but I don't have a strong feeling either way.